### PR TITLE
Liqun/OpenTelemetry

### PR DESCRIPTION
**Pull request by liqul**
Add opentelemetry tracing framework to taskweaver

### CODE REVIEW

File: website/docs/tracing.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+After running the docker image, you can access the Jaeger frontend at `http://localhost:16686`.
+Now, when you run TaskWeaver, issue a task, and access the Jaeger frontend, you can see the traces of the task execution.
+On the left side panel, you can select the Service dropdown to filter the traces by the service name.
+The service name of TaskWeaver is `taskweaver.opentelemetry.tracer`.
+
+## How to customize tracing
+
+The instrumentation of TaskWeaver is done by the OpenTelemetry Python SDK.
+So, if you want to customize the tracing, you need to modify the TaskWeaver code.
```
Comment (ShilinHe) [inline website/docs/tracing.md:64]: Can users add tracing to the plugin? guess they do not have the needs to instrument TaskWeaver core, but for plugins, it is highly possible

File: website/sidebars.js [hunk 2]
```diff
@@ -81,7 +81,8 @@ const sidebars = {
     'code_verification',
     'code_execution',
     'FAQ',
-    'cli_only'
+    'cli_only',
+    'training',
```
Comment (ShilinHe) [inline website/sidebars.js:85]: training? or tracing?

File: taskweaver/module/tracing.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+from taskweaver.config.module_config import ModuleConfig
+
+
+class TracingConfig(ModuleConfig):
+    def _configure(self):
+        self._set_name("tracing")
+        self.enabled = self._get_bool("enabled", False)
+        self.endpoint = self._get_str("endpoint", "http://127.0.0.1:4318/v1/traces")
+        self.service_name = self._get_str("service_name", "taskweaver.opentelemetry.tracer")
```
Comment (ShilinHe) [inline taskweaver/module/tracing.py:13]: feel like the defult name is too long when displaying the Jaeger UI (component name is then difficult to read), may consider simplify it

File: taskweaver/module/tracing.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        self.exporter = self._get_str("exporter", "otlp")
```
Comment (ShilinHe) [inline taskweaver/module/tracing.py:14]: what else value can this be configured?

Comment (ShilinHe) [reply taskweaver/module/tracing.py:14]: I see the "console", so the trace would be printed directly to console? then Jaeger is not needed to display the traces? sounds like this is more lightweight than the "otlp" option

File: website/docs/tracing.md (hunk 1 above)
Comment (liqul) [reply website/docs/tracing.md:64]: It is a bit complicated to consider the generated code in tracing. I think it possible, but we do not quite understand the motivation. 

The current tracing is to let users know the time consumption of each part and potentially the costs.

File: taskweaver/module/tracing.py (hunk 3 above)
Comment (liqul) [reply taskweaver/module/tracing.py:13]: need to have both 'taskweaver' and 'opentelemetry'.

File: taskweaver/module/tracing.py (hunk 4 above)
Comment (liqul) [reply taskweaver/module/tracing.py:14]: Console is purely for debugging and will mess up the output. So it is not a real option. This configure is not supposed to be exposed to users.

File: website/sidebars.js (hunk 2 above)
Comment (liqul) [reply website/sidebars.js]: fixed

File: website/docs/tracing.md (hunk 1 above)
Comment (ShilinHe) [reply website/docs/tracing.md:64]: My thought was: plugin execution might take some time... we can address this later if someone has the needs.

File: taskweaver/module/tracing.py (hunk 3 above)
Comment (ShilinHe) [reply taskweaver/module/tracing.py:13]: for example, using "tw" and "otel" are not allowed? in Jaeger, its readability is not very good.

Comment (liqul) [reply taskweaver/module/tracing.py:13]: taskweaver.optl.tracer? tw is not so good :(

Comment (ShilinHe) [reply taskweaver/module/tracing.py:13]: okay, taskweaver.otel.tracer? otel is the official abbreviation for OpenTelemetry

Comment (liqul) [reply taskweaver/module/tracing.py]: OpenTelemetry Protocol (OTLP)

**Review by ShilinHe — APPROVED** [review]

**Pull request merged**